### PR TITLE
Improve silently

### PR DESCRIPTION
**Pull request by mvz**
This does two things:
* Make `Silencer.silently` correctly restore the original situation.
* Removes use of `Silencer.silently` where it seems no longer necessary.

### CODE REVIEW

**Comment by mvz** [comment]
My future ideal would be to pass IO down from `bin/reek` and never call `Kernel.puts` or `Kernel.warn`.

File: lib/reek/ast/node.rb [hunk 1]
```diff
@@ -2,9 +2,7 @@
 
 require_relative '../cli/silencer'
 
-Reek::CLI::Silencer.silently do
-  require 'parser'
-end
+require 'parser'
```
Comment (troessner) [inline lib/reek/ast/node.rb]: So that problem with the parser gem was fixed upstream? How did you notice?

File: lib/reek/cli/silencer.rb [hunk 2]
```diff
@@ -8,17 +8,20 @@ module CLI
...
+        old_stdout = $stdout
+
         $VERBOSE = false
         $stderr = StringIO.new
         $stdout = StringIO.new
         yield
       ensure
         $VERBOSE = old_verbose
-        $stderr = STDERR
-        $stdout = STDOUT
```
Comment (troessner) [inline lib/reek/cli/silencer.rb]: Nice catch.

File: spec/reek/cli/silencer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    end
+
+    it 'restores output on $stdout after the block' do
+      expect do
+        described_class.silently { puts 'Hi!' }
+        puts 'there!'
+      end.to output("there!\n").to_stdout
+    end
+
+    it 'blocks output on $stderr after the block' do
```
Comment (troessner) [inline spec/reek/cli/silencer_spec.rb]: Shouldn't this also read "restores output ...." like above?

File: spec/reek/configuration/excluded_paths_spec.rb [hunk 4]
```diff
@@ -12,9 +12,7 @@
       let(:paths) { [smelly_source_dir, file_as_path] }
 
       it 'raises an error if one of the given paths is a file' do
-        Reek::CLI::Silencer.silently do
-          expect { exclusions.add(paths) }.to raise_error(Reek::Errors::ConfigFileError)
-        end
```
Comment (troessner) [inline spec/reek/configuration/excluded_paths_spec.rb]: Interesting. I can't even recall why we had that there.

File: spec/reek/tree_dresser_spec.rb [hunk 5]
```diff
@@ -1,7 +1,4 @@
-require_lib 'reek/cli/silencer'
-Reek::CLI::Silencer.silently do
-  require 'parser/ruby22'
-end
+require 'parser/ruby22'
```
Comment (troessner) [inline spec/reek/tree_dresser_spec.rb]: How about making this consistent to above and use ruby25 in regard to the good old boyscout rule? ;)

File: lib/reek/ast/node.rb (hunk 1 above)
Comment (mvz) [reply lib/reek/ast/node.rb]: I removed the silencer and noticed no extra output.

Comment (mvz) [reply lib/reek/ast/node.rb]: This whole PR came from my hunch on what's going on in #1357.

File: spec/reek/configuration/excluded_paths_spec.rb (hunk 4 above)
Comment (mvz) [reply spec/reek/configuration/excluded_paths_spec.rb]: The method emitted a warning and called `exit` at some point.

**Comment by mvz** [comment]
I processed your comments, @troessner.

**Pull request merged**